pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,208 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer_test
+
+import (
+	"fmt"
+	"os/exec"
+	"testing"
+
+	utilstest "github.com/kinvolk/inspektor-gadget/pkg/gadgets/internal/test"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+func TestCreateTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil,
+		func(types.Event) {})
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(tracer.Stop)
+}
+
+func TestStopTracerIdempotent(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil,
+		func(types.Event) {})
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+
+	// Check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func createTracer(
+	t *testing.T, config *tracer.Config, callback func(types.Event),
+) *tracer.Tracer {
+	t.Helper()
+
+	tracer, err := tracer.NewTracer(config, nil, callback)
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	return tracer
+}
+
+func TestTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	manyArgs := []string{}
+	// 19 is DEFAULT_MAXARGS - 1 (-1 because args[0] is on the first position).
+	for i := 0; i < 19; i++ {
+		manyArgs = append(manyArgs, "/dev/null")
+	}
+
+	// Function to generate an event used most of the times.
+	// Returns pid of executed process.
+	generateEvent := func() (int, error) {
+		cmd := exec.Command("/bin/cat", "/dev/null")
+		if err := cmd.Run(); err != nil {
+			return 0, fmt.Errorf("running command: %w", err)
+		}
+
+		return cmd.Process.Pid, nil
+	}
+
+	tests := map[string]utilstest.Test[types.Event, int]{
+		"captures_all_events_without_configured_filters": {
```

Comment:
I know I previously suggested what's in there, but now I look at it and I feel it could be further rephrased.
```suggestion
		"captures_all_events_with_no_filters_configured": {
```